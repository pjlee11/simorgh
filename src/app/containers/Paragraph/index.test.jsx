import React from 'react';
import { shouldShallowMatchSnapshot } from '../../helpers/tests/testHelpers';
import ParagraphContainer from './index';

const fragmentBlock = (text, attributes = []) => ({
  type: 'fragment',
  model: {
    text,
    attributes,
  },
});

const inlineLinkBlock = (locator, blocks) => ({
  type: 'urlLink',
  model: {
    locator,
    blocks,
  },
});

const inlineLink = inlineLinkBlock('/bbc-test', [
  fragmentBlock('Some text'),
  fragmentBlock(' for the ', ['bold']),
  fragmentBlock(' link!', ['italic']),
]);

const blocks = [fragmentBlock('This is some text.', ['bold']), inlineLink];

describe('ParagraphContainer', () => {
  shouldShallowMatchSnapshot(
    'should render correctly',
    <ParagraphContainer blocks={blocks} />,
  );
});
