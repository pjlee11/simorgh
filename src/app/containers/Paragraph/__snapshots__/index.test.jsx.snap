// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParagraphContainer should render correctly 1`] = `
<Paragraph>
  <Blocks
    blocks={
      Array [
        Object {
          "model": Object {
            "attributes": Array [
              "bold",
            ],
            "text": "This is some text.",
          },
          "type": "fragment",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "Some text",
                },
                "type": "fragment",
              },
              Object {
                "model": Object {
                  "attributes": Array [
                    "bold",
                  ],
                  "text": " for the ",
                },
                "type": "fragment",
              },
              Object {
                "model": Object {
                  "attributes": Array [
                    "italic",
                  ],
                  "text": " link!",
                },
                "type": "fragment",
              },
            ],
            "locator": "/bbc-test",
          },
          "type": "urlLink",
        },
      ]
    }
    componentsToRender={
      Object {
        "fragment": [Function],
        "urlLink": [Function],
      }
    }
  />
</Paragraph>
`;
